stacks/serverless/dovetail-cdn-arranger.yml

```diff
@@ -0,0 +1,133 @@
+      # TODO: tag-specific expirations
+      LifecycleConfiguration:
+        Rules:
+          - ExpirationInDays: 30
+            Status: Enabled
+      Tags:
+        - Key: Project
+          Value: Dovetail
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: prx:cloudformation:stack-name
+          Value: !Ref AWS::StackName
+        - Key: prx:cloudformation:stack-id
+          Value: !Ref AWS::StackId
+  DovetailCdnArrangerBinaryLayer:
+    Type: AWS::Lambda::LayerVersion
+    Properties:
+      CompatibleRuntimes:
+        - nodejs14.x
+        - nodejs12.x
+      Content:
+        S3Bucket: prx-tech
+        S3Key: archives/ffmpeg-4.3.1-lambda-layer.zip
+      Description: ffmpeg binaries for dovetail-cdn-arranger
+      LicenseInfo: MIT
+  DovetailCdnArrangerExecutionIAMRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'lambda.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      Path: '/'
+      Policies:
+        - PolicyName: DynamoDBWritePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'dynamodb:DescribeTable'
+                  - 'dynamodb:GetItem'
+                  - 'dynamodb:PutItem'
+                Resource:
+                  - !Ref DynamodbTableArn
+        - PolicyName: S3WritePolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:ListBucket'
+                Resource:
+                  - !GetAtt DovetailCdnArrangerWorkspaceBucket.Arn
+              - Effect: Allow
+                Action:
+                  - 's3:GetObject'
+                  - 's3:PutObject'
+                  - 's3:PutObjectTagging'
+                Resource:
+                  - !Sub ${DovetailCdnArrangerWorkspaceBucket.Arn}/*
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy'
+      Tags:
+        - Key: Project
+          Value: Dovetail
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: 'prx:cloudformation:stack-name'
+          Value: !Ref AWS::StackName
+        - Key: 'prx:cloudformation:stack-id'
+          Value: !Ref AWS::StackId
+  DovetailCdnArrangerFunction:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Code:
+        S3Bucket: !Ref CodeS3Bucket
+        S3Key: lambda/PRX-dovetail-cdn-arranger.zip
+        S3ObjectVersion: !Ref CodeS3ObjectVersion
+      Description: Dovetail CDN Arranger
+      Environment:
+        Variables:
+          DDB_TABLE: !Ref DynamodbTableArn
+          DOVETAIL_HOST: !Ref DovetailHost
+          DOVETAIL_TOKEN: !Ref DovetailToken
```

Comment:
Without knowing specifically what `host` and `token` are here, I feel like they may be the type of values that would be good to have version history on, which we do kind of get with Parameter Store, but not in a way that is compatible with our rollbacks. In the absence of something more similar to `aws-secrets` for Lambdas, I'm wondering if we move these to the JSON file.

Reply:
Yeah, +1 to having a better way to version these.  But for now, I'm fine with ParameterStore translating to ENVs, just to get this out there.  These will hardly ever change, and in the edge stitch-lambda, I actually have them hardcoded in github.